common/persistence/nosql/nosqlplugin/cassandra/workflow_utils_test.go

```diff
@@ -1349,6 +1366,1125 @@ func TestUpdateChildExecutionInfos(t *testing.T) {
 	}
 }
 
+func TestResetTimerInfos(t *testing.T) {
+	ts, err := time.Parse(time.RFC3339, "2023-12-12T22:08:41Z")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tests := []struct {
+		desc       string
+		shardID    int
+		domainID   string
+		workflowID string
+		runID      string
+		timerInfos map[string]*persistence.TimerInfo
+		// expectations
+		wantQueries []string
+	}{
+		{
+			desc:       "ok",
+			shardID:    1000,
+			domainID:   "domain1",
+			workflowID: "workflow1",
+			runID:      "runid1",
+			timerInfos: map[string]*persistence.TimerInfo{
+				"timer1": {
+					Version:    1,
+					TimerID:    "timer1",
+					StartedID:  2,
+					ExpiryTime: ts.UTC(),
+					TaskStatus: 1,
+				},
+			},
+			wantQueries: []string{
+				`UPDATE executions SET timer_map = map[` +
+					`timer1:map[expiry_time:2023-12-12 22:08:41 +0000 UTC started_id:2 task_id:1 timer_id:timer1 version:1]` +
+					`] WHERE ` +
+					`shard_id = 1000 and type = 1 and domain_id = domain1 and workflow_id = workflow1 and ` +
+					`run_id = runid1 and visibility_ts = 946684800000 and task_id = -10 `,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.desc, func(t *testing.T) {
+			batch := &fakeBatch{}
+
+			err := resetTimerInfos(batch, tc.shardID, tc.domainID, tc.workflowID, tc.runID, tc.timerInfos)
+			if err != nil {
+				t.Fatalf("resetTimerInfos() error = %v", err)
+			}
+
+			if diff := cmp.Diff(tc.wantQueries, batch.queries); diff != "" {
+				t.Fatalf("Query mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestUpdateTimerInfos(t *testing.T) {
+	ts, err := time.Parse(time.RFC3339, "2023-12-19T22:08:41Z")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		desc        string
+		shardID     int
+		domainID    string
+		workflowID  string
+		runID       string
+		timerInfos  map[string]*persistence.TimerInfo
+		deleteInfos []string
+		// expectations
+		wantQueries []string
+	}{
+		{
+			desc:       "update and delete timer infos",
+			shardID:    1000,
+			domainID:   "domain1",
+			workflowID: "workflow1",
+			runID:      "runid1",
+			timerInfos: map[string]*persistence.TimerInfo{
+				"timer1": {
+					TimerID:    "timer1",
+					Version:    1,
+					StartedID:  2,
+					ExpiryTime: ts.UTC(),
+					TaskStatus: 1,
+				},
+			},
+			deleteInfos: []string{"timer2"},
+			wantQueries: []string{
+				`UPDATE executions SET timer_map[ timer1 ] = {` +
```

Comment:
I'm a bit unsure, but for me it looks like these tests are testing both updateTimerInfos and custom implementation of FakeBatcher.
I'm not sure we can guarantee that this queries are what will be executed by Cassandra, since this is a simplified stub.

Reply:
That's right and that's actually the scope of unit tests. Basically the function under test takes a `gocql.Batch` interface and calls it with certain parameters. Test is validating this part. What is out of scope for these unit tests is the actual queries generated by the concrete implementation of 3rd party cql package which should be a concern of integration test